internal/cli/internal/command/dig/dig.go

```diff
@@ -238,10 +194,51 @@ func run(ctx context.Context) error {
 	return nil
 }
 
-// ResolverForOrg takes a connection to the wireguard agent and an organization
+// roundTrip a DNS request across a "TCP" socket; we'd just use miekg/dns's Client, but I don't think it promises to
+// work over our weird UDS TCP proxy.
+func roundTrip(conn net.Conn, m *dns.Msg) (*dns.Msg, error) {
+	m.Id = dns.Id()
+	m.Compress = true
+
+	buf, err := m.Pack()
+	if err != nil {
+		return nil, fmt.Errorf("dns round trip: %w", err)
```

Comment:
this isn't a dns round trip error; I wouldn't decorate my errors in a helper function like `roundTrip` but at the caller.

Reply:
@azazeal  Yeah indeed. But if i's, not a round trip error, what decoration would give it in the main function?

Reply:
just don't decorate them and bubble them up. all of these are errors due to writing to the connection - it'll be obvious where they're coming from